transport/x/grpc/outbound.go

```diff
@@ -162,20 +157,37 @@ func (o *Outbound) stop() error {
 	return nil
 }
 
-func errorToGRPCError(ctx context.Context, request *transport.Request, start time.Time, err error) error {
-	deadline, _ := ctx.Deadline()
-	ttl := deadline.Sub(start)
-	switch grpc.Code(err) {
-	case codes.DeadlineExceeded:
-		return errors.ClientTimeoutError(request.Service, request.Procedure, ttl)
-	case codes.Unimplemented, codes.InvalidArgument, codes.NotFound:
-		return errors.RemoteBadRequestError(grpc.ErrorDesc(err))
-	case codes.Canceled, codes.AlreadyExists, codes.PermissionDenied,
-		codes.Unauthenticated, codes.ResourceExhausted, codes.FailedPrecondition,
-		codes.Aborted, codes.OutOfRange, codes.Internal,
-		codes.Unavailable, codes.DataLoss, codes.Unknown:
-		fallthrough
-	default:
-		return errors.RemoteUnexpectedError(grpc.ErrorDesc(err))
+func invokeErrorToYARPCError(err error, responseMD metadata.MD) error {
+	if err == nil {
+		return nil
+	}
+	if yarpcerrors.IsYARPCError(err) {
+		return err
+	}
+	status, ok := status.FromError(err)
+	// if not a yarpc error or grpc error, just return the error
```

Comment:
This comment states that you want to return the error as-is if it's *not* a YARPC error but you're returning it in the opposite case.